### DISCUSSION TITLE
Middleware::Sidekiq in Rspec Testing

### DISCUSSION THREAD

**Discussion by SuguramK** [sidekiq/sidekiq#6169] [Q&A] [answered]
Hi @mperham  is it compulsory to clear the Middleware after running a rspec? 

Will the below code in a spec file can impact the follow up spec files ? Do we need to clear the Sidekiq::Testing.server_middleware  exclusively even if we have self.use_transactional_tests = true. Is there any document for this scenario ?

```
Sidekiq::Testing.server_middleware do |chain|
  chain.add Middleware::Sidekiq::Server::SomeJobs
end
```

**Comment by mperham** [comment, marked as answer]
Yes, `Sidekiq::Testing.server_middleware` is a global variable so you have to keep it tidy.

**Reply by SuguramK** [reply]
Thanks @mperham Your help is much appreciated. Thanks once again 😃
